### ISSUE TITLE
_sslverify.problemsFromTransport should be deprecated

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2874]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2874|
|Type|enhancement|
|Created|2007-11-06 10:45:52Z|
|Branch|https://github.com/twisted/twisted/tree/delete-problemsFromTransport-2874|

It's obsolete, but we should first remove its use in amp.

<details><summary>Searchable metadata</summary>

```
trac-id__2874 2874
type__enhancement enhancement
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_delete_problemsFromTransport_2874 branches/delete-problemsFromTransport-2874
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1194345952000000 1194345952000000
changetime__1217187167000000 1217187167000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24195]) Branching to 'delete-problemsFromTransport-2874'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24196]) remove usage of problemsFromTransport from amp and then remove problemsFromTransport and all related support code

refs [#2874](https://github.com/twisted/twisted/issues/2874)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

It's removed.  There's one small change in behavior.  AMP `IBoxReceiver`s now get the underlying OpenSSL exception passed to their `stopReceivingBoxes` method instead of an `OpenSSLVerifyError` (which was never exposed publically).

The underlying OpenSSL exception has more useful information in it, so this seems like an improvement to me.  The most likely way this could break someone's program is if they imported `OpenSSLVerifyError` from the private `twisted.internet._sslverify` and did type checking on exceptions.  That exception is no longer defined, so they'll get an `ImportError`.

Build results are here:

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/delete-problemsFromTransport-2874&num_builds=1

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [24403]) Merge delete-problemsFromTransport-2874

Author: exarkun
Reviewer: therve
Fixes: [#2874](https://github.com/twisted/twisted/issues/2874)

Remove the private `problemsFromTransport` API which exposed OpenSSL errors.
This API was redundant with `IProtocol.connectionLost`, which is called
with a `Failure` wrapped around an `OpenSSL.SSL.Error` if there is a problem
with SSL which results in the connection being lost.

AMP is also changed to use the failure passed to `connectionLost` rather
than `problemsFromTransport`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
